### PR TITLE
[NP-1614] Add MinMaxCapability

### PR DESCRIPTION
**Pull request by KernelDeimos**
Putting this up so parts can be reviewed ahead of time. I need to discuss part of it with Ruby to figure out the missing piece (dealing with chained UCJ checks for agent junctions)

### CODE REVIEW

**Comment by KernelDeimos** [comment]
This is... missing stuff? I lost some code. Trying to find it now.

File: src/foam/nanos/crunch/MinMaxCapability.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        for ( CapabilityCapabilityJunction ccJunction : ccJunctions ) {
+          Capability cap = (Capability) ccJunction.findSourceId(x);
+          if ( ! cap.getEnabled() ) continue;
+
+          UserCapabilityJunction ucJunction =
+            crunchService.getJunction(x, ccJunction.getTargetId());
+          if ( ucJunction == null ) continue;
+
+          switch ( ucJunction.getStatus() ) {
+            case GRANTED:
```
Comment (noodlemoodle) [inline src/foam/nanos/crunch/MinMaxCapability.js:67]: I think this switch need to check for the following as well
GRACE_PERIOD == GRANTED
APPROVED == PENDING

File: src/foam/nanos/crunch/MinMaxCapability.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+foam.CLASS({
```
Comment (jlhughes) [inline src/foam/nanos/crunch/MinMaxCapability.js:7]: license header please, and on others.

**Review by jlhughes — COMMENTED** [review]
Assume 'max' support is forthcoming.

File: src/foam/nanos/crunch/MinMaxCapability.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    'foam.dao.DAO',
+    'java.util.List',
+    'static foam.mlang.MLang.*'
+  ],
+
+  properties: [
+    {
+      name: 'min',
+      class: 'Int',
+      value: 1,
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/MinMaxCapability.js]: No trailing commas here and elsewhere.

Comment (KernelDeimos) [reply src/foam/nanos/crunch/MinMaxCapability.js]: oops, not trying to break the styleguide it's just really hard to break the habit

**Pull request merged**